cvat-ui/src/components/annotation-page/standard-workspace/controls-side-bar/tools-control.tsx

```diff
@@ -1197,173 +1196,69 @@ export class ToolsControlComponent extends React.PureComponent<Props, State> {
-                                        Number.isFinite(+value) &&
-                                        +value >= +min &&
-                                        +value <= +max &&
-                                        !(+value % +step)
-                                    ) {
-                                        return {
-                                            ...acc,
-                                            [attributeSpec.id as number]: `${value}`,
-                                        };
-                                    }
-
-                                    return acc;
-                                }
-                                case 'text': {
-                                    return {
-                                        ...acc,
-                                        [attributeSpec.id as number]: `${value}`,
-                                    };
-                                }
-                                case 'select':
-                                case 'radio': {
-                                    if (attributeSpec.values.includes(value)) {
-                                        return {
-                                            ...acc,
-                                            [attributeSpec.id as number]: value,
-                                        };
-                                    }
-                                    return acc;
-                                }
-                                case 'checkbox':
-                                    return {
-                                        ...acc,
-                                        [attributeSpec.id as number]: `${value}`.toLowerCase() === 'true' ? 'true' : 'false',
-                                    };
-                                default:
-                                    return acc;
-                            }
-                        }, {} as Record<number, string>);
+                        return Object.fromEntries(attributes.map((a) => [a.spec_id, a.value]));
                     }
 
                     try {
                         this.setState({ mode: 'detection', fetching: true });
 
-                        // The function call endpoint doesn't support the cleanup and conv_mask_to_poly parameters.
-                        const { cleanup, conv_mask_to_poly: convMaskToPoly, ...restOfBody } = body;
+                        // The function call endpoint doesn't support the cleanup parameter.
+                        const { cleanup, ...restOfBody } = body;
 
                         const result = await core.lambda.call(jobInstance.taskId, model, {
                             ...restOfBody, frame, job: jobInstance.id,
-                        }) as DetectedShapes;
+                        }) as DetectorResults;
+
+                        const tagStates = result.tags.map((tag) => {
+                            const jobLabel = jobInstance.labels
```

Comment:
`!` finding corresponding label is not really guaranteed, as it may be added after opening the job, thus, passing validation on the server and failing here

Reply:
but maybe that is not a thing we should be worried

Reply:
I don't think that's possible. If the client doesn't know about a label, it will not include it in the mapping, so the server should not return any shapes with that label.